Implements support for PHP 5.3

soebbing opened the pull request:
Hi KuiKui,

your wish is my command. Can this be merged?

Have fun,
Hendrik

KuiKui commented on src/KuiKui/MemcacheServiceProvider/ServiceProvider.php:
```diff
@@ -22,15 +22,16 @@ class ServiceProvider implements ServiceProviderInterface
     {
-        $app['memcache'] = $app->share(function() use ($app) {
-            $memcacheClass = $this->getMemcacheClass($app);
+        $that = $this;
```
Can you rename `$that` to `$provider` and add a inline comment to explain this trick ? :sunglasses:

KuiKui commented:
Can you add a new test in [ServiceProvider.php](https://github.com/KuiKui/MemcacheServiceProvider/blob/master/tests/units/KuiKui/MemcacheServiceProvider/ServiceProvider.php#L12) :
```php
    public function testRegisterWrapperSelection()
    {
        $app = new Application();
        $provider = new MemcacheServiceProvider\ServiceProvider();

        $provider->register($app);

        $this->object($app['memcache'])
            ->isInstanceOf('\KuiKui\MemcacheServiceProvider\AbstractWrapper')
        ;
    }
```

Or a better one :wink:

soebbing commented:
The pattern is fairly common in JavaScript, where closures are used a bit longer than in PHP. There you just don't have to make the copy of $this explicitly available in the closure as you have to in PHP.

I changed the name of the var, added a comment and your test.

KuiKui commented:
Thanks !

KuiKui commented:
Fichtre!
We have to fix tests :sweat:
I check this tomorrow.

KuiKui commented:
Can you merge `master` in your feature branch (I've updated Travis config) ?

soebbing commented:
Master was merged into a new feature branch, see other pull request.
